subsys/testsuite/ztest/include/ztest_assert.h

```diff
@@ -168,6 +168,19 @@ static inline void z_zassert(int cond,
 	zassert((void *)(a) == (void *)(b), #a " not equal to  " #b, \
 		msg, ##__VA_ARGS__)
 
+/**
+ * @brief Assert that @a is within @a b with delta @a d
+ *
+ * @param a Value to compare
+ * @param b Value to compare
+ * @param d Delta
+ * @param msg Optional message to print if the assertion fails
+ */
+#define zassert_within(a, b, d, msg, ...)			     \
+	zassert(((a) > ((b) - (d))) && ((a) < ((b) + (d))),	     \
+		#a " not within  " #b " +/- delta " #d,		     \
```

Comment:
Double space after "within". Intentional?
Perhaps the word "delta" could be omitted in the message.